DragaliaAPI/DragaliaAPI/Features/StorySkip/StorySkipService.cs

```diff
@@ -0,0 +1,304 @@
+using System.Collections.Frozen;
+using DragaliaAPI.Database;
+using DragaliaAPI.Database.Entities;
+using DragaliaAPI.Database.Repositories;
+using DragaliaAPI.Features.Fort;
+using DragaliaAPI.Shared.Definitions.Enums;
+using DragaliaAPI.Shared.Features.StorySkip;
+using DragaliaAPI.Shared.MasterAsset;
+using DragaliaAPI.Shared.MasterAsset.Models;
+using DragaliaAPI.Shared.MasterAsset.Models.Story;
+using static DragaliaAPI.Shared.Features.StorySkip.StorySkipRewards;
+
+namespace DragaliaAPI.Features.StorySkip;
+
+public class StorySkipService(
+    ApiContext apiContext,
+    IFortRepository fortRepository,
+    ILogger<StorySkipService> logger,
+    IQuestRepository questRepository,
+    IStoryRepository storyRepository,
+    IUserDataRepository userDataRepository
+)
+{
+    public void IncreaseFortLevels(long viewerId)
+    {
+        Dictionary<FortPlants, FortConfig> fortConfigs = StorySkipRewards.FortConfigs;
+        IQueryable<DbFortBuild> userForts = fortRepository.Builds.Where(x =>
+            x.ViewerId == viewerId
+        );
```

Comment:
Not sure if you realize, but every time you do a `Any()`, `Count()` or iterate through with `foreach` on an `IQueryable` you are executing a database query. I would suggest realizing the results of the query before you do your processing:

```suggestion
        List<DbFortBuild> userForts = await fortRepository.Builds.Where(x =>
            x.ViewerId == viewerId
        ).ToListAsync();
```

Reply:
I did not realize this. I changed these objects to Lists.